maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -955,27 +957,20 @@ protected final Toolchain getToolchain()
 
         if ( jdkToolchain != null )
         {
-            // Maven 3.3.1 has plugin execution scoped Toolchain Support
             try
             {
-                Method getToolchainsMethod =
-                    toolchainManager.getClass().getMethod( "getToolchains", MavenSession.class, String.class,
-                        Map.class );
-
-                @SuppressWarnings( "unchecked" )
+                Method getToolchainsMethod = ReflectionUtils.getMethod( getToolchainManager(), "getToolchains" );
                 List<Toolchain> tcs =
-                    (List<Toolchain>) getToolchainsMethod.invoke( toolchainManager, getSession(), "jdk",
-                        jdkToolchain );
-
+                    (List<Toolchain>) ReflectionUtils.invokeMethodWithArray( getToolchainManager(),
+                        getToolchainsMethod, getSession(), "jdk" );
                 if ( tcs != null && !tcs.isEmpty() )
```

Comment:
After the method `tryGetMethod ` returns `getToolchainsMethod` is not null then the `invokeMethodWithArray` would not throw reflection exception and the null check of for `tcs` does not make sense because the `List` would not be null, see the [Javadoc](http://maven.apache.org/ref/3.3.9/maven-core/apidocs/org/apache/maven/toolchain/ToolchainManager.html#getToolchains(org.apache.maven.execution.MavenSession,%20java.lang.String,%20java.util.Map)).